data/json/mapgen/house/bungalow09.json

```
@@ -0,0 +1,94 @@
+[
+  {
+    "type": "mapgen",
+    "om_terrain": "bungalow09",
+    "weight": 250,
+    "method": "json",
+    "object": {
+      "palettes": [
+        "standard_domestic_palette",
+        "standard_domestic_lino_kitchen",
+        "standard_domestic_lino_bathroom",
+        "standard_domestic_landscaping_palette"
+      ],
+      "fill_ter": "t_floor",
+      "rows": [
+        "..........-----p........",
+        "...[....???```???.[...[.",
+        "..!!!!!!?```````?!!!!!..",
+        "..###o###````₸K₸##o###..",
+        "..#Db  ¤####*oo##r  R#..",
+        "..o   @@|Ly))HHH|rh R#..",
+        "..#d  @@|x ))))H|P  R#..",
+        "..#d    |x ))l)H|||¶|#..",
+        "..#||||+|V ))l)HT   y#..",
+        "..#8BB|    ))))s   h o..",
+        "..#8((+           ffh#..",
+        "..oS(Q|     h    hff #..",
+        "..#t(w|b   rrrRR   h o..",
+        "..#|||||+|+||||||(JJJ#..",
+        ".^#IIID  | ddD|Z+(((3#^.",
+        "..#      |    |W+(((O#..",
+        "..o     d|  @@|||F((1o..",
+        "..# @@  d|  @@|6+(((5#..",
+        "..#¤@@¤ y|y  ¤|6+((42#..",
+        "..##o##o###o######*o##..",
+        "........jjj....`````₸?..",
+        "................?``₸K?..",
+        "........[.......??????..",
+        "........................"
+      ],
+      "terrain": {
+        "?": "t_railing_v",
+        "K": "t_concrete",
+        "₸": "t_concrete",
+        "(": "t_linoleum_gray",
+        "W": "t_linoleum_gray",
+        "Z": "t_linoleum_gray",
+        ")": "t_carpet_yellow",
+        "H": "t_carpet_yellow",
+        "l": "t_carpet_yellow",
+        "s": "t_carpet_yellow"
+      },
+      "place_loot": [ { "item": "laptop", "x": 17, "y": 5, "chance": 100 }, { "item": "television", "x": 9, "y": 7, "chance": 100 } ]
+    }
+  },
+  {
+    "type": "mapgen",
+    "om_terrain": "bungalow09_roof",
+    "weight": 250,
+    "method": "json",
+    "object": {
+      "palettes": [ "roof_palette" ],
+      "fill_ter": "t_shingle_flat_roof",
+      "rows": [
+        "````````````````````````",
+        "````````````````````````",
+        "````````````````````````",
+        "``-------```````------``",
+        "``3.....---------....|``",
+        "``3..................|``",
+        "``3..................|``",
+        "``3..................|``",
+        "``3...N..............|``",
+        "``3..................|``",
+        "``3.............ssss.|``",
+        "``3.............ssss.|``",
+        "``3..................|``",
+        "``3..................|``",
+        "``5..................5``",
+        "``3..................|``",
+        "``3..................|``",
+        "``3..................|``",
+        "``3..................|``",
+        "``22222222222222222222``",
+        "````````````````````````",
+        "````````````````````````",
+        "````````````````````````",
+        "````````````````````````"
+      ],
+      "terrain": { "`": "t_open_air", ".": "t_shingle_flat_roof" },
+      "furniture": { "s": "f_solar_unit" }
```

```suggestion
      "furniture": { "s": "f_solar_unit" }
```
I'd prefer if you made a nested chunk for rooftops of a solar setup that's some solar units and whatnot. Then have those chunks have a low chance on lots of rooftops

---

I've thought about something similar, but in my opinion, having a generic rooftop "chunk" that contains yet other chunks including solar, enemy encounters, etc. etc. would be better. That way, you'd only have to add the generic residential rooftop chunk overlay on each house, and whenever you wanted to add/remove/tweak the chunks for, say, all residential rooftop spawns, you don't edit each house json file individually -- rather, you'd tweak the "master" chunk that fills in each of the house rooftops. That would be beyond the scope of this specific PR though.